src/overlays/actors/ovl_En_Trt/z_en_trt.c

```diff
@@ -22,176 +103,1714 @@ const ActorInit En_Trt_InitVars = {
+    this->sleepSoundTimer = 10;
+    this->actor.textId = 0;
+    this->actionFunc = EnTrt_IdleSleeping;
+    this->blinkFunc = EnTrt_NodOff;
+}
+
+s32 EnTrt_TestEndInteraction(EnTrt* this, GlobalContext* globalCtx, Input* input) {
+    if (CHECK_BTN_ALL(input->press.button, BTN_B)) {
+        EnTrt_EndInteraction(globalCtx, this);
+        return true;
+    }
+    return false;
+}
+
+s32 EnTrt_TestCancelOption(EnTrt* this, GlobalContext* globalCtx, Input* input) {
+    if (CHECK_BTN_ALL(input->press.button, BTN_B)) {
+        this->actionFunc = this->tmpActionFunc;
+        func_80151938(globalCtx, EnTrt_GetItemTextId(this));
+        return true;
+    }
+    return false;
+}
+
+void EnTrt_SetupStartShopping(GlobalContext* globalCtx, EnTrt* this, u8 skipHello) {
+    func_8011552C(globalCtx, 0x10);
+    if (!skipHello) {
+        this->actionFunc = EnTrt_Hello;
+    } else {
+        EnTrt_StartShopping(globalCtx, this);
+    }
+}
+
+void EnTrt_StartShopping(GlobalContext* globalCtx, EnTrt* this) {
+    func_80151938(globalCtx, 0x83E);
+    func_8011552C(globalCtx, 6);
+    this->stickLeftPrompt.isEnabled = false;
+    this->stickRightPrompt.isEnabled = true;
+    this->actionFunc = EnTrt_FaceShopkeeper;
+}
+
+void EnTrt_UpdateJoystickInputState(GlobalContext* globalCtx, EnTrt* this) {
+    s8 stickX = CONTROLLER1(globalCtx)->rel.stick_x;
+    s8 stickY = CONTROLLER1(globalCtx)->rel.stick_y;
+
+    if (this->stickAccumX == 0) {
+        if (stickX > 30 || stickX < -30) {
+            this->stickAccumX = stickX;
+        }
+    } else if (stickX <= 30 && stickX >= -30) {
+        this->stickAccumX = 0;
+    } else {
+        if ((this->stickAccumX * stickX) < 0) { // Stick has swapped directions
+            this->stickAccumX = stickX;
+        } else {
+            this->stickAccumX += stickX;
+            if (this->stickAccumX > 2000) {
+                this->stickAccumX = 2000;
+            } else if (this->stickAccumX < -2000) {
+                this->stickAccumX = -2000;
+            }
+        }
+    }
+    if (this->stickAccumY == 0) {
+        if (stickY > 30 || stickY < -30) {
+            this->stickAccumY = stickY;
+        }
+    } else {
+        if (stickY <= 30 && stickY >= -30) {
+            this->stickAccumY = 0;
+        } else {
+            if ((this->stickAccumY * stickY) < 0) { // Stick has swapped directions
+                this->stickAccumY = stickY;
+            } else {
+                this->stickAccumY += stickY;
+                if (this->stickAccumY > 2000) {
+                    this->stickAccumY = 2000;
+                } else if (this->stickAccumY < -2000) {
+                    this->stickAccumY = -2000;
+                }
+            }
+        }
+    }
+}
+
+u8 EnTrt_SetCursorIndexFromNeutral(EnTrt* this, u8 shelfOffset) {
+    if (this->items[shelfOffset] != NULL) {
+        return shelfOffset;
+    }
+    return CURSOR_INVALID;
+}
+
+void EnTrt_Hello(EnTrt* this, GlobalContext* globalCtx) {
```

Comment:
Would EnTrt_Greeting make more sense than Hello? Just a thought

Reply:
I'm fine with either or. Main reason I went with Hello was consistency between all the shopkeepers both here in MM and OOT (where they have a Hello State).

Reply:
Consistency makes sense to me